cli/src/pcluster/validators/ebs_validators.py

```diff
@@ -31,7 +31,7 @@ class EbsVolumeTypeSizeValidator(Validator):
     The volume sizes of st1 and sc1 range from 500 GiB - 16 TiB(16384 GiB)
     """
 
-    def _validate(self, volume_type: Param, volume_size: Param):
+    def _validate(self, volume_type: Resource.Param, volume_size: Resource.Param):
```

Comment:
Same question as above. When executing `ebs.validate()`, the arguments are `str`/`int` instead of `Resource.Param`. This problem is not dected because of the absence of unit tests to test the validation of a whole Resource. 
I am fine with either of the fixings below: 
1. Change the type hint to `str`/`int`, remove all '.value's in validators and remove all `Resource.Param(xxx)` in unit tests for validators
2. Change every `_register_validators()` to pass in `Resource.Param` instead of the `str`/`int`

Reply:
You are totally right; see my answer above.
We decided to adopt solution 2.

Reply:
Actually, on a second thought we switched to solution 1. The reason is that this option goes in the same direction of this patch, i.e. removing the explicit use of the `Param` class.